Modules/_ctypes/_ctypes.c

```diff
@@ -2613,12 +2648,11 @@ static int PyCData_NewGetBuffer(PyObject *myself, Py_buffer *view, int flags)
     view->format = dict->format ? dict->format : "B";
     view->ndim = dict->ndim;
     view->shape = dict->shape;
-    view->itemsize = self->b_size;
-    if (view->itemsize) {
-        for (i = 0; i < view->ndim; ++i) {
-            view->itemsize /= dict->shape[i];
-        }
+    item_dict = PyCData_item_stgdict(dict);
+    if (item_dict == NULL) {
+        return NULL;
```

Comment:
Should return 0

Reply:
It seems from the [PEP](https://legacy.python.org/dev/peps/pep-3118) that for `getbufferproc` the error status is -1

Reply:
Good catch, thanks

Reply:
Fixed